### ISSUE TITLE
 “gradle check” ERROR：undefined symbol: cblas_dgemv

### ISSUE THREAD

**Issue by Sun-shan** [broadinstitute/hail#565]
Hi, Today I updated the new verion of hail ，and try to re-build it, but I encountered some issues. 
      
（1） “gradle check” ERROR ：/tmp/jniloader803664626041947143netlib-native_system-linux-x86_64.so: undefined symbol: cblas_dgemv
----------------
The “~/hail $ gradle installDist” went successfully，but when I do "~/hail $ gradle check",it got some error:

........
Gradle suite > Gradle test > org.broadinstitute.hail.methods.LinearRegressionSuite.testNonNumericPheno PASSED
Running test: Test method testWithImportFam(org.broadinstitute.hail.methods.LinearRegressionSuite)
/opt/BioDir/jdk/jdk1.8.0_91/bin/java: symbol lookup error: /tmp/jniloader803664626041947143netlib-native_system-linux-x86_64.so: undefined symbol: cblas_dgemv
:test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> Process 'Gradle Test Executor 1' finished with non-zero exit value 127

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 5 mins 6.833 secs

**Comment by danking** [comment]
This means that hail (or something on which hail depends) is trying to call into BLAS. BLAS is a Fortran library and is often shipped with C bindings. The symbol `cblas_dgemv` is a C function. Your machine is likely missing `libcblas`.

Can you post the output of these commands:

 - `nm -g /tmp/jniloader803664626041947143netlib-native_system-linux-x86_64.so` (this may say that the file doesn't exist, in which case skip the next command
 - `objdump -TC /tmp/jniloader803664626041947143netlib-native_system-linux-x86_64.so`

Can you also answer these questions:

 - What distribution are you using?
 - What version of that distribution do you have?
 - What package management tool do you use?

If you're on Ubuntu, can you tell me what version of `libatlas-dev` you have installed?

**Comment by danking** [comment]
And in the off chance that you're using ArchLinux, can you try installing `openblas-lapack` from the AUR?

**Comment by Sun-shan** [comment]
hello，when I run the commands:
（1）
root (master) hail $  nm -g /tmp/jniloader803664626041947143netlib-native_system-linux-x86_64.so
  nm: /tmp/jniloader803664626041947143netlib-native_system-linux-x86_64.so: no symbols
（2）
objdump -TC /tmp/jniloader803664626041947143netlib-native_system-linux-x86_64.so 
There are many lines，I attach it in a file as follows 
[netlib_info.txt](https://github.com/broadinstitute/hail/files/412441/netlib_info.txt)

Our OS distribution is CentOS 7.1 ，there are tools such as yum and so on

**Comment by danking** [comment]
Can you post the results of
```
yum info atlas-devel
```

If `atlas-devel` is not installed, can you try installing it?
```
yum install atlas-devel
```

**Comment by Sun-shan** [comment]
（1）yum info atlas-devel
root yum.repos.d $ yum info atlas-devel
Loaded plugins: fastestmirror, langpacks
base                                                                                                                                         | 3.6 kB  00:00:00     
extras                                                                                                                                       | 3.4 kB  00:00:00     
updates                                                                                                                                      | 3.4 kB  00:00:00     
(1/4): base/7/x86_64/group_gz                                                                                                                | 155 kB  00:00:00     
(2/4): extras/7/x86_64/primary_db                                                                                                            | 160 kB  00:00:00     
(3/4): base/7/x86_64/primary_db                                                                                                              | 5.3 MB  00:00:09     
(4/4): updates/7/x86_64/primary_db                                                                                                           | 6.5 MB  00:00:32     
Loading mirror speeds from cached hostfile
 * base: mirror.bit.edu.cn
 * epel: mirrors.neusoft.edu.cn
 * extras: mirrors.tuna.tsinghua.edu.cn
 * updates: mirrors.tuna.tsinghua.edu.cn
Available Packages
Name        : atlas-devel
Arch        : i686
Version     : 3.10.1
Release     : 10.el7
Size        : 1.5 M
Repo        : base/7/x86_64
Summary     : Development libraries for ATLAS
URL         : http://math-atlas.sourceforge.net/
License     : BSD
Description : This package contains the libraries and headers for development
            : with ATLAS (Automatically Tuned Linear Algebra Software).

Name        : atlas-devel
Arch        : x86_64
Version     : 3.10.1
Release     : 10.el7
Size        : 1.5 M
Repo        : base/7/x86_64
Summary     : Development libraries for ATLAS
URL         : http://math-atlas.sourceforge.net/
License     : BSD
Description : This package contains the libraries and headers for development
            : with ATLAS (Automatically Tuned Linear Algebra Software).
（2）I installed the “atlas-devel” , 
---------
root yum.repos.d $ yum install atlas-devel
Loaded plugins: fastestmirror, langpacks
Loading mirror speeds from cached hostfile
 * base: mirror.bit.edu.cn
 * epel: mirrors.neusoft.edu.cn
 * extras: mirror.bit.edu.cn
 * updates: mirror.bit.edu.cn
Resolving Dependencies
--> Running transaction check
---> Package atlas-devel.x86_64 0:3.10.1-10.el7 will be installed
--> Processing Dependency: atlas = 3.10.1-10.el7 for package: atlas-devel-3.10.1-10.el7.x86_64
.............

Installed:
  atlas-devel.x86_64 0:3.10.1-10.el7                                                                                                                                

Dependency Installed:
  atlas.x86_64 0:3.10.1-10.el7                                                                                                                                      

Complete!
-----------------
######**but when I excute the "gradle check --info" ，the error still appeared.**
------------------
/opt/BioDir/jdk/jdk1.8.0_91/bin/java: symbol lookup error: /tmp/jniloader7277009897699512423netlib-native_system-linux-x86_64.so: undefined symbol: cblas_dgemv

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> Process 'Gradle Test Executor 1' finished with non-zero exit value 127

* Try:
Run with --stacktrace option to get the stack trace. Run with --debug option to get more log output.
----------  
#######The output info was collected in the file as follow:
[gradle_check_info1.txt](https://github.com/broadinstitute/hail/files/417544/gradle_check_info1.txt)

**Comment by danking** [comment]
Hi @Sun-shan,

I am unsure what is wrong.

I tried to replicate your environment as follows:

 - I downloaded the CentOS 7.2 1511 ["everything ISO"](http://isoredirect.centos.org/centos/7/isos/x86_64/CentOS-7-x86_64-Everything-1511.iso)
 - On a VM, I installed CentOS using that iso
 - I downloaded the Gradle "Binary distribution" from the [Gradle website](https://gradle.org/gradle-download/)
 - I downloaded a zip file of the hail repository from github
 - In the hail directory, I issued `gradle installDist`, which succeeded
 - In the hail directory, I issued `gradle check`, which succeeded except for the five tests that require PLINK or R

I did not see any undefined symbol errors. Unfortunately, further debugging your environment is outside of the scope of this project.

The only remaining recommendation I can give is to use the (slow) reference implementations of BLAS functions. To use the reference implementations, run the following command instead of `gradle check`:

```bash
gradle -Dcom.github.fommil.netlib.BLAS=com.github.fommil.netlib.NativeRefBLAS check
```

---

The following details about the VM may be helpful if you attempt to modify your system.

```
[dking@cg-router1 hail-master]$ rpm --query centos-release
centos-release-7-2.1511.el7.centos.2.10.x86_64
```
```
[dking@cg-router1 hail-master]$ hostnamectl
   Static hostname: cg-router1.broadinstitute.org
         Icon name: computer-vm
           Chassis: vm
        Machine ID: 0d856e1616ee4961bfc1b76c6ec420a1
           Boot ID: 1fc0d1ffc3d24218a81ea8fc5abd9776
    Virtualization: kvm
  Operating System: CentOS Linux 7 (Core)
       CPE OS Name: cpe:/o:centos:centos:7
            Kernel: Linux 3.10.0-327.el7.x86_64
      Architecture: x86-64
```

The output of `yum list installed` is in [installed-packages.txt](https://github.com/broadinstitute/hail/files/422887/installed-packages.txt)
